src/composables/transactionList.ts

```diff
@@ -1,29 +1,31 @@
-import { orderBy, uniqBy } from 'lodash-es';
+import { uniqBy } from 'lodash-es';
 import { ref, watch } from 'vue';
-import { Encoded, Tag } from '@aeternity/aepp-sdk';
+import { Encoded } from '@aeternity/aepp-sdk';
 import type {
   ITransaction,
   ITransactionsState,
   IAccountTransactionsState,
   IDefaultComposableOptions,
 } from '@/types';
 import {
+  PROTOCOL_AETERNITY,
+  PROTOCOL_BITCOIN,
   TRANSACTIONS_LOCAL_STORAGE_KEY,
   TX_DIRECTION,
 } from '@/constants';
 import {
-  fetchJson,
   getLocalStorageItem,
   setLocalStorageItem,
 } from '@/utils';
-import JsonBig from '@/lib/json-big';
-import { AEX9_TRANSFER_EVENT } from '@/protocols/aeternity/config';
-import { useAeNetworkSettings } from '@/protocols/aeternity/composables';
 
+import { ProtocolAdapterFactory } from '@/lib/ProtocolAdapterFactory';
+// TS-ignore
+import { useAeTransactionList } from '@/protocols/aeternity/composables/aeTransactionsList';
 import { useAccounts } from './accounts';
-import { useMiddleware } from './middleware';
 import { useAeSdk } from './aeSdk';
 
+const btcAdapter = ProtocolAdapterFactory.getAdapter(PROTOCOL_BITCOIN);
```

Comment:
The goal for us is to avoid any use of data, variables, components related to one specific protocol in the global scope. This composable is global, so what it should is:
1. Get currently used protocols
2. Get adapters for those protocols
3. Run the correct adapter method to fetch the transactions.

So the adapters should have correct methods for all cases.